core/arch/arm/arm.mk

```diff
@@ -166,7 +168,17 @@ ta_arm32-platform-cflags += $(arm32-platform-cflags)
 ta_arm32-platform-cflags += $(platform-cflags-optimization)
 ta_arm32-platform-cflags += $(platform-cflags-debug-info)
 ta_arm32-platform-cflags += -fpic
-ta_arm32-platform-cflags += $(arm32-platform-cflags-generic)
+
+# Thumb mode doesn't support function graph tracing due to missing
+# frame pointer support required to trace function call chain. So
+# rather compile in ARM mode if function tracing is enabled.
+ifeq ($(CFG_TA_FTRACE_SUPPORT),y)
+ta_arm32-platform-cflags += $(arm32-platform-cflags-generic-arm) \
```

Comment:
Unless `CFG_TA_MCOUNT`  is set too, this isn't very useful.

Reply:
There is another case if `CFG_LIBS_MCOUNT` is enabled. Also, it might be the case if user enable `-pg` for a particular file via corresponding `sub.mk`. So I would prefer to keep it tied to `CFG_TA_FTRACE_SUPPORT=y` only.

Reply:
OK